### PR TITLE
Quick pre/2.6 patches

### PR DESCRIPTION
**Pull request by daquinteroflex**
- Again improve the develop CLI command
- Add the codespell precommit hook.
- Try to automate the sync-action to get triggered on any pre/* deployment, and to get out the existing branch name. Need to test this
- Fix minor pre/2.5 conflicts that got badly resolved.
- Add sphinx emoji to begin sprinkling around in a reasonable fashion
- CLI commands for notebooks updates and remote builds, improved relevant docs

### CODE REVIEW

File: docs/api/heat/boundary_conditions.rst [hunk 1]
```diff
@@ -1,10 +1,10 @@
 .. currentmodule:: tidy3d
 
 Thermal Boundary Conditions
-===============================
+--------------------
```
Comment (tylerflex) [inline docs/api/heat/boundary_conditions.rst:4]: I thought these underlines must be longer than the title?

File: docs/api/heat/index.rst [hunk 2]
```diff
@@ -1,5 +1,5 @@
-HEAT
-=======
+HEAT |:fire:|
```
Comment (tylerflex) [inline docs/api/heat/index.rst:1]: nice haha

File: docs/api/heat/boundary_conditions.rst (hunk 1 above)
Comment (daquinteroflex) [reply docs/api/heat/boundary_conditions.rst:4]: Yeah all done now

**Comment by daquinteroflex** [comment]
So this has some CLI commands that should help the notebooks team so thought maybe handy to merge this so they can test the new dev flow more easily

File: tidy3d/web/cli/develop.py [hunk 3]
```diff
@@ -349,31 +351,45 @@ def commit_repository(repository_path, commit_message):
...
 def test_base_tidy3d(args=None):
-    """Verifies and builds the documentation."""
-    # Runs the documentation build from the poetry environment
-    # TODO update generic path management.
     echo_and_run_subprocess(["poetry", "run", "pytest", "-rA", "tests"])
     return 0
 
 
+@develop.command(name="test-notebooks", help="Tests the tidy3d notebooks.")
+def test_notebooks_tidy3d(args=None):
```
Comment (tylerflex) [inline tidy3d/web/cli/develop.py:387]: any way (here or in a future PR) to add specific notebook names to run only? say I want to run the `ParameterSweep.ipynb` notebook (or maybe others) I can do `tidy3d develop test-notebooks ParameterSweep`? or something

**Review by tylerflex — APPROVED** [review]
Looks good. I already took a look earlier. Just one comment regarding some utilities to select specific notebooks to re-run in the CLI.

File: tidy3d/web/cli/develop.py (hunk 3 above)
Comment (daquinteroflex) [reply tidy3d/web/cli/develop.py:387]: Yeah I think so through CLI arguments, I'll include it in my list for a future PR

**Pull request merged**